.github/workflows/test_ci_cases.yaml

```diff
@@ -0,0 +1,46 @@
+name: CI tests
+
+on: 
+  push:
+    branches: [ "main" ]
+
+  pull_request:
+    branches: [ "main" ]
+
+jobs: 
+  CI-tests:
+    runs-on: self-hosted
+    steps: 
+
+      - name: Checkout latest enclave-cc source code
+        if: ${{ github.event_name == 'push' }}
+        uses: actions/checkout@v3
+        with:
+          ref: main
+
+      - name: Checkout PR source code
+        if: ${{ github.event_name == 'pull_request' }}
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
my experience is that one checkout for both PRs and main push is enough

Reply:
If I don't specify the lines of `with: ref: xxxx`, the checkout action will pull the source code from repo by default, rather than from PR commits.

Reply:
> I'd want to see the results from the self-hosted runner before merging this PR.

https://github.com/confidential-containers/enclave-cc/actions/runs/4080955181/jobs/7035496977

Reply:
> If I don't specify the lines of `with: ref: xxxx`, the checkout action will pull the source code from repo by default, rather than from PR commits.

only one `uses: actions/checkout@v3` should be enough.

> > I'd want to see the results from the self-hosted runner before merging this PR.
> 
> https://github.com/confidential-containers/enclave-cc/actions/runs/4080955181/jobs/7035496977

Sorry I was confused by "CI tests" which I thought refers to our original checks. Can we move this functionality under `enclave-cc e2e test`?

Reply:
Do you mean move these CI tests to `https://github.com/confidential-containers/enclave-cc/blob/main/.github/workflows/e2e.yaml`, and only run CI tests after every PR is merged?

Reply:
we want CI tests for every PR but given the limitations we have with the payload images through the operator based installation, I was thinking could we just reuse `e2e.yaml` based flow and create a matrix workflow for HW (runs-on self-hosted) and SW (runs-on ubuntu-xxx).